packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -4547,6 +4547,10 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
       config.controlsNodes = _properties.controlsNodes;
     }
 
+    if (_properties.inputType != null) {
+      config.inputType = _properties.inputType!;
+    }
```

Comment:
Should `CustomPaint` support this too? Not sure if anyone would use custom-paint for a whole text input though.

Reply:
filed https://github.com/flutter/flutter/issues/166374